src/segments/pulumi.go

```diff
@@ -0,0 +1,237 @@
+package segments
+
+import (
+	"crypto/sha1"
+	"encoding/hex"
+	"encoding/json"
+	"fmt"
+	"github.com/jandedobbeleer/oh-my-posh/src/platform"
+	"github.com/jandedobbeleer/oh-my-posh/src/properties"
+	"gopkg.in/yaml.v3"
+	"time"
+)
+
+const (
+	PulumiUser properties.Property = "pulumi_user"
+	PulumiURL  properties.Property = "pulumi_url"
+)
+
+type Pulumi struct {
+	props properties.Properties
+	env   platform.Environment
+	Stack string
+
+	SpecFile
+	Backend
+}
+
+type SpecFile struct {
+	FileName string
+	Kind     string
+	Path     string
+}
+
+type Backend struct {
+	URL  string
+	User string
+}
+
+type PulumiStackLSJson struct {
+	Name          string `json:"name"`
+	Current       bool   `json:"current"`
+	ResourceCount int    `json:"resourceCount"`
+	LastUpdate    string `json:"lastUpdate"`
+}
+
+type PulumiAboutJS struct {
+	Backend *PulumiBackendJS `json:"backend"`
+}
+
+type PulumiBackendJS struct {
+	URL  string `json:"url"`
+	User string `json:"user"`
+}
+
+type PulumiFileSpec struct {
+	Name string `yaml:"name" json:"name"`
+}
+
+type PulumiWorkSpaceFileSpec struct {
+	Stack string `json:"stack"`
+}
+
+func (p *Pulumi) Template() string {
+	return "{{ .Stack }}{{if .User }} :: {{ .User }}@{{ end }}{{ if .URL }}{{ .URL }}{{ end }}"
+}
+
+func (p *Pulumi) Init(props properties.Properties, env platform.Environment) {
+	p.props = props
+	p.env = env
+}
+
+func (p *Pulumi) Enabled() bool {
+	err := p.getPulumiFile()
+	if err != nil {
+		p.env.Error(err)
+		return false
+	}
+	if p.props.GetBool(PulumiUser, false) && p.props.GetBool(PulumiURL, false) {
+		return p.getPulumiStackName() && p.getPulumiAbout()
+
+	} else {
+		return p.getPulumiStackName()
+	}
+}
+
+func (p *Pulumi) getPulumiStackName() bool {
+
+	project, err := p.getProjectName()
+	if err != nil {
+		p.env.Error(err)
+		return false
+	}
+
+	sha1HexString, err := p.sha1HexString(p.SpecFile.Path)
+
+	if err != nil {
+		p.env.Error(err)
+		return false
+	}
+
+	stackNameFile := project + "-" + sha1HexString + "-" + "workspace.json"
+
+	homedir := p.env.Home()
+
+	workspaceCacheFile := homedir + "/.pulumi/workspaces/" + stackNameFile
+	if p.env.HasFiles(workspaceCacheFile) {
+		stackNameFileContent := p.env.FileContent(workspaceCacheFile)
+		var pulumiWorkspaceSpec PulumiWorkSpaceFileSpec
+		err = json.Unmarshal([]byte(stackNameFileContent), &pulumiWorkspaceSpec)
+		if err != nil {
+			p.env.Error(fmt.Errorf("PULUMI STACK FILE DECODE ERROR"))
+			return false
+		}
+		p.Stack = pulumiWorkspaceSpec.Stack
+		return true
+
+	}
```

Comment:
```suggestion
	stackNameFileContent := p.env.FileContent(workspaceCacheFile)
	if len(stackNameFileContent) != 0 {
		var pulumiWorkspaceSpec PulumiWorkSpaceFileSpec
		err = json.Unmarshal([]byte(stackNameFileContent), &pulumiWorkspaceSpec)
		if err != nil {
			p.env.Error(fmt.Errorf("PULUMI STACK FILE DECODE ERROR"))
			return false
		}
		p.Stack = pulumiWorkspaceSpec.Stack
		return true

	}
```

`HasFiles` checks if the directory has trigger files, according to `display_mode` property, like [there](https://ohmyposh.dev/docs/segments/helm#properties), this should work as intended.